### PR TITLE
feat(useDraggable): support `capture` option

### LINKED ISSUE #2724: useDraggable support trigger in bubbling phase
### Clear and concise description of the problem

Currently, useDraggable trigger event in capturing phase, but somtimes there are draggable element nested in the draggable element, if we drag the one inside, it will affect the one outside.

### Suggested solution

provide a option helping user choose weather trigger in bubbling phase

### Alternative

_No response_

### Additional context

_No response_

### Validations

- [X] Follow our [Code of Conduct](https://github.com/vueuse/vueuse/blob/main/CODE_OF_CONDUCT.md)
- [X] Read the [Contributing Guidelines](https://github.com/vueuse/vueuse/blob/main/CONTRIBUTING.md).
- [X] Read the [docs](https://vueuse.org/guide).
- [X] Check that there isn't already an issue that request the same feature to avoid creating a duplicate.

### PR DESCRIPTION
**Pull request by faga295**
closed #2724 

### Additional context

---

### What is the purpose of this pull request? 

- [ ] Bug fix
- [x] New Feature
- [ ] Documentation update
- [ ] Other

### Before submitting the PR, please make sure you do the following

- [x] Read the [Contributing Guidelines](https://github.com/vueuse/vueuse/blob/main/CONTRIBUTING.md).
- [x] Read the [Pull Request Guidelines](https://github.com/vueuse/vueuse/blob/main/packages/guidelines.md).
- [x] Check that there isn't already a PR that solves the problem the same way to avoid creating a duplicate.
- [x] Provide a description in this PR that addresses **what** the PR is solving, or reference the issue that it solves (e.g. `fixes #123`).
- [x] Ideally, include relevant tests that fail without this PR but pass with it.

### CODE REVIEW

File: packages/core/useDraggable/index.ts [hunk 1]
```diff
@@ -135,9 +141,10 @@ export function useDraggable(target: MaybeComputedRef<HTMLElement | SVGElement |
   }
 
   if (isClient) {
-    useEventListener(draggingHandle, 'pointerdown', start, true)
-    useEventListener(draggingElement, 'pointermove', move, true)
-    useEventListener(draggingElement, 'pointerup', end, true)
+    const useCapture = resolveUnref(options?.useCapture ?? true)
+    useEventListener(draggingHandle, 'pointerdown', start, useCapture)
+    useEventListener(draggingElement, 'pointermove', move, useCapture)
+    useEventListener(draggingElement, 'pointerup', end, useCapture)
```
Comment (vaakian) [inline packages/core/useDraggable/index.ts:147]:
```suggestion
    let cleanups: Fn[] = []
    watch(resolveRef(options.useCapture), (useCapture = true) => {
      cleanups.forEach(cleanup => cleanup())
      cleanups = [
        useEventListener(draggingHandle, 'pointerdown', start, useCapture),
        useEventListener(draggingElement, 'pointermove', move, useCapture),
        useEventListener(draggingElement, 'pointerup', end, useCapture),
      ]
    }, { immediate: true })
```
eventListeners should be re-registered when `useCapture` ref value changes.

**Review by vaakian — CHANGES_REQUESTED** [review]

File: packages/core/useDraggable/index.ts (hunk 1 above)
Comment (faga295) [reply packages/core/useDraggable/index.ts]: Thank you for your suggestion!

File: packages/core/useDraggable/index.ts [hunk 2]
```diff
@@ -135,9 +141,16 @@ export function useDraggable(target: MaybeComputedRef<HTMLElement | SVGElement |
...
+    let cleanups: Fn[] = []
+    const useCapture = computed(() => resolveUnref(options?.useCapture ?? true))
+    watch(useCapture, (useCapture = true) => {
+      cleanups.forEach(cleanup => cleanup())
+      cleanups = [
+        useEventListener(draggingHandle, 'pointerdown', start, useCapture),
+        useEventListener(draggingElement, 'pointermove', move, useCapture),
+        useEventListener(draggingElement, 'pointerup', end, useCapture),
+      ]
+    }, { immediate: true })
```
Comment (antfu) [inline packages/core/useDraggable/index.ts:153]: I think we could make `useEventListener` able to accept ref for the options, so we don't need the workaround here.

**Review by antfu — CHANGES_REQUESTED** [review]
Please refactor using https://github.com/vueuse/vueuse/commit/103d6c03fa5bb5f3415337b7626dc334175a0e50

Thanks

**Comment by faga295** [comment]
Has refactored, please review again

**Comment by artel1992** [comment]
I would recommend adding this code in case someone decides to call 'open' twice, once with 'capture' and once without it:
     if (hasOwn(_options, 'capture'))
            input.capture = _options.capture!
        else
            input.removeAttribute('capture')

**Pull request merged**